core/server/services/auth/api-key/admin.js

```diff
@@ -52,6 +46,38 @@ const authenticate = (req, res, next) => {
         }));
     }
 
+    JWT_OPTIONS.maxAge = '5m';
+    return authenticateWithToken(req, res, next, token);
+};
+
+const authenticateWithUrl = (req, res, next) => {
+    const token = _extractTokenFromUrl(req.originalUrl);
+    if (!token) {
+        return next(new common.errors.UnauthorizedError({
+            message: common.i18n.t('errors.middleware.auth.invalidTokenWithMessage', {message: 'No token found in URL'}),
+            code: 'INVALID_JWT'
+        }));
+    }
+    // CASE: Scheduler publish URLs can have long maxAge but controllerd by expiry and neverBefore
+    delete JWT_OPTIONS.maxAge;
```

Comment:
The: `JWT_OPTIONS.maxAge = '5m';` and `delete JWT_OPTIONS.maxAge;` seems a little odd here (they mutate a "global" object which somehow is affecting the output of the function they are calling). I'd expect rather having `5m` always set as a default and only ignore that value. Maybe passing `options` object as the last parameter to `authenticateWithToken` where the object would look like:
```
{
	token: token,
    JWT_OPTIONS: _.omit(JWT_OPTIONS, 'maxAge')
}
```
:thinking:

Reply:
➕ seems odd to be modifying the global constant that is supposed to represent default options

Reply:
Yeah agreed not an ideal way to set this up, I'll refactor this closer to as @gargol suggested 👍

Reply:
Updated